qa/L0_trt_dla/dla_client.py

```diff
@@ -0,0 +1,133 @@
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import numpy as np
+import sys
+from PIL import Image
+
+import tritonclient.http as httpclient
+from tritonclient.utils import InferenceServerException
+
+
+def preprocess(img, dtype, c, h, w, scaling):
+    """
+    Pre-process an image to meet the size and type
+    requirements specified by the parameters.
+    """
+
+    sample_img = img.convert('RGB')
+    resized_img = sample_img.resize((w, h), Image.BILINEAR)
+    resized = np.array(resized_img)
+
+    if resized.ndim == 2:
+        resized = resized[:, :, np.newaxis]
+
+    typed = resized.astype(dtype)
+
+    if scaling == 'INCEPTION':
+        scaled = (typed / 127.5) - 1
+    elif scaling == 'VGG':
+        if c == 1:
+            scaled = typed - np.asarray((128,), dtype=dtype)
+        else:
+            scaled = typed - np.asarray((123, 117, 104), dtype=dtype)
+    else:
+        scaled = typed
+
+    ordered = np.transpose(scaled, (2, 0, 1))
+
+    return ordered
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-v',
+                        '--verbose',
+                        action="store_true",
+                        required=False,
+                        default=False,
+                        help='Enable verbose output')
+    parser.add_argument('-u',
+                        '--url',
+                        type=str,
+                        required=False,
+                        default='localhost:8000',
+                        help='Inference server URL. Default is localhost:8000.')
+    parser.add_argument(
+        '-s',
+        '--scaling',
+        type=str,
+        choices=['NONE', 'INCEPTION', 'VGG'],
+        required=False,
+        default='NONE',
+        help='Type of scaling to apply to image pixels. Default is NONE.')
+    parser.add_argument('image_filename',
+                        type=str,
+                        default=None,
+                        help='Input image.')
+
+    FLAGS = parser.parse_args()
+    try:
+        triton_client = httpclient.InferenceServerClient(
+            url=FLAGS.url, verbose=FLAGS.verbose)
+    except Exception as e:
+        print("channel creation failed: " + str(e))
+        sys.exit(1)
+
+    model_name = "resnet50_plan"
```

Comment:
Since the Python script is targeting for this particular model and input in the test, it doesn't need to be constructed like the client examples, a unit test style makes more sense as the arguments above are actually determined and thus the `preprocess()` helper function can also be simplified.

Reply:
Done